Q1. What are the three measures of central tendency?

solution: Mean, Median and Mode are the 3 measures of central tendancy

Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

Mean: it is the average of all the datapoints. sum(datapoints)/len(datapoints). mean is not a good process to use when there are outliers in the data.
Median: for calculating median we need to sort the datapoints. if numbers of datapoints are even. take average of middile 2 elements and whatever answer is coming that is considered as median. if number of elements are odd then middle element is the median. it is used when you want to deal with outliers present in the data.
Mode: Which datapoint is occuring maximum number of times or frequency of datapoints is mode.

Q3.Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [1]:
import numpy as np
data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
mean1=np.mean(data)
print("Mean of height data is:",mean1)

Mean of height data is: 177.01875


In [2]:
import numpy as np
data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
mean1=sum(data)/len(data)
print("Mean of height data is:",mean1)

Mean of height data is: 177.01875


In [4]:
import numpy as np
data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
med=np.median(data)
print("Median of height data is:",med)

Median of height data is: 177.0


data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
sorted_data=[172.5,175,175,175,176,176.2,176.5,177,177,177,178,178,178.2,178.9,179,180]
since numbers of elements are even therefore taking 2 middle elements that is 177,177.
taking average of 2 middle elments: 177+177/2=354/2= 177
therefore 177 is the medain

In [10]:
import numpy as np
data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
mode=np.unique(data)
print("Mode of height data is:",mode)

Mode of height data is: [172.5 175.  176.  176.2 176.5 177.  178.  178.2 178.9 179.  180. ]


mode of the data is: [177.0,178.0]

Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [12]:
import numpy as np
import math
data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
variance=np.var(data)
std_deviation=np.sqrt(variance)
print("Varinace of height data is:",variance)
print("Standard Deviation of height data is:",std_deviation)

Varinace of height data is: 3.199023437500001
Standard Deviation of height data is: 1.7885814036548633


variance=[(178-177)^2+(177-177)^2+(176-177)^2+(177-177)^2+(178.2-177)^2+(178-177)^2+(175-177)^2+(179-177)^2
          +(180-177)^2+(175-177)^2+(178.9-177)^2+(176.2-177)^2+(177-177)^2+(172.5-177)^2+(178-177)^2+(176.5-177)^2]/16

variance=1+0+1+0+1.44+1+4+4+9+4+3.61+0.64+0+20.25+1+0.25/16
variance=51.19/16
variance=3.199
standard_deviation=1.788

Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.

soultion:Measures of dispersion, such as range, variance, and standard deviation, are used to describe how much the data points in a dataset vary or spread out from the central tendency (usually represented by the mean). These measures provide insights into the degree of variability or dispersion within the data and can help understand the overall distribution of the dataset.

Range: represents the difference between the largest and smallest values in the dataset. gives rough idea of spread of data but sensitive to outliers.

Variance: takes into account individual distances of each data point from mean. It is average of squared differences between each data point and mean. Variance is always non-negative, and higher values indicate more significant dispersion.

Standard Deviation: square root of the variance and is the most widely used measure of dispersion. It provides measure of dispersion in original units of the data, making it more interpretable. A higher standard deviation indicates more spread or variability in the dataset.

Example:
following dataset representing daily temperatures recorded for past 10 days:

{ 28, 30, 29, 25, 31, 26, 27, 32, 29, 28 }

Step 1: mean of the dataset.
Mean = (28 + 30 + 29 + 25 + 31 + 26 + 27 + 32 + 29 + 28) / 10 = 28.5

Step 2: Calculate range.
Range = Maximum value - Minimum value = 32 - 25 = 7

Step 3: Calculate variance.
Variance = [(28 - 28.5)^2 + (30 - 28.5)^2 + (29 - 28.5)^2 + (25 - 28.5)^2 + (31 - 28.5)^2 + (26 - 28.5)^2 + (27 - 28.5)^2 + (32 - 28.5)^2 + (29 - 28.5)^2 + (28 - 28.5)^2] / 10
Variance = 5.25

Step 4: Calculate standard deviation.
Standard Deviation = √5.25 = 2.29

In this example, range is 7, variance is approximately 5.25, and standard deviation is approximately 2.29. standard deviation being relatively low suggests that the temperatures are not significantly spread out from the mean, indicating a relatively stable climate during the recorded period.

Q6. What is a Venn diagram?

solution: A Venn diagram is graphical representation used to show relationships between different sets of items. It was introduced by John Venn in the late 19th century as a way to visualize set theory concepts. Venn diagrams use overlapping circles or other shapes to depict the common and distinct elements among different sets.

main parts of a Venn diagram are:

Circles: Each circle represents a set. The items or elements belonging to that set are enclosed within the circle.

Overlapping regions: When two or more circles overlap, the shared area represents the elements that are common to the sets involved in the overlap.

The size of the circles and their overlap can vary based on the number of elements in each set and the relationships between the sets. The more overlap there is between circles, the more elements two or more sets have in common.

Venn diagrams are commonly used in various fields, including mathematics, logic, statistics, and data analysis, to illustrate set operations, logical relationships, and the intersection between different categories.

Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A intersection B
(ii) A ⋃ B

Solution: (i) A intersection B =(2,6)
(ii) A ⋃ B=(0,2,3,4,5,6,7,8,10)

Q8. What do you understand about skewness in data?
Soultion: Data is not symmetrical. That is all sides are not equal. There is a bending in the data. it quantifies the extent to which a dataset deviates from being symmetrical or bell-shaped

Q9. If a data is right skewed then what will be the position of median with respect to mean?
Solution: occurs when the tail of the distribution is stretched out to the right, indicating that the data has a few large values that pull the mean in that direction. The majority of the data is concentrated towards the lower values.
mean is typically larger than the median

Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?
Solution: Correlation is a standardized measure of the linear relationship between two variables. Unlike covariance, correlation provides a more interpretable value that ranges between -1 and 1. A correlation of 1 indicates a perfect positive linear relationship, -1 indicates a perfect negative linear relationship, and 0 indicates no linear relationship between the variables. Correlation allows for a more direct comparison between different pairs of variables because it is not affected by the scale of the variables.
Covariance measures the degree to which two variables change together. It shows whether an increase or decrease in one variable is associated with a similar change in the other variable. A positive covariance indicates that as one variable increases, the other tends to increase as well, and vice versa for decreases. Conversely, a negative covariance indicates that as one variable increases, the other tends to decrease, and vice versa.

Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.
solution: summation of xi/n where xi=datapoint n=sample size 
dataset: [28, 34, 25, 29, 31]
Now, values from the dataset:
x_bar=28+34+25+29+31/5
x_bar=147/5
x_bar=29.4
So, the sample mean of this dataset is 29.4. 

Q12. For a normal distribution data what is the relationship between its measure of central tendency?
Solution: Mean (μ): The mean is the arithmetic average of all the data points in the distribution. For a perfectly symmetrical normal distribution, the mean is equal to the median and mode. This is one of the defining characteristics of a normal distribution.

Median: The median is the middle value of the data when arranged in ascending or descending order. For a perfectly symmetrical normal distribution, the median is also equal to the mean and mode.

Mode: The mode is the value that occurs with the highest frequency in the data. In a normal distribution, the mode is also equal to the mean and median in a symmetrical distribution.

So, for a normal distribution, the mean, median, and mode are all equal when the distribution is perfectly symmetrical. 

Q13.How is covariance different from correlation?
Solution: The key difference between covariance and correlation is that covariance provides information about the direction of the relationship between two variables (positive or negative), but it does not quantify the strength of the relationship. Correlation, on the other hand, not only gives the direction but also provides a standardized measure of the strength, making it more interpretable and comparable across datasets with different units and scales.






Q14. Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.
solution: 
Outliers can have a significant impact on measures of central tendency and dispersion in a dataset. Measures of central tendency, such as the mean, median, and mode, are used to represent the typical or central value of the data, while measures of dispersion, like the range, variance, and standard deviation, indicate how spread out the data points are.

Measures of Central Tendency:

Mean: The mean is particularly sensitive to outliers because it takes into account every data point. Even a single extreme outlier can significantly affect the mean, pulling it away from the center of the data.
Median: The median is less affected by outliers. It represents the middle value when the data is arranged in ascending or descending order. Outliers only impact the median when they lie outside the middle portion of the data.
Mode: The mode represents the most frequently occurring value in the dataset. Outliers that are far from the rest of the data may not affect the mode.
Measures of Dispersion:

Range: The range is the difference between the maximum and minimum values in the dataset. Outliers can increase the range if they are extreme values, leading to a larger spread.
Variance and Standard Deviation: Both variance and standard deviation quantify the average deviation of data points from the mean. Outliers increase the variance and standard deviation because they can be significantly distant from the mean, widening the spread.
Example:
Consider a dataset of exam scores for a math class:

65, 70, 72, 75, 78, 80, 85, 95, 100, 120

Here, the dataset contains an outlier, 120, which is significantly higher than the rest of the scores. Let's see how it affects the measures of central tendency and dispersion:

Central Tendency:

Mean: (65 + 70 + 72 + 75 + 78 + 80 + 85 + 95 + 100 + 120) / 10 = 80
With the outlier included, the mean is pulled towards the higher value of 80.

Median: Since there are 10 data points, the median is the average of the 5th and 6th values: (78 + 80) / 2 = 79
The median is less affected by the outlier and remains close to the center of the data.

Mode: There is no mode in this dataset since all values occur only once.
Dispersion:

Range: Range = 120 - 65 = 55
The range is increased significantly due to the presence of the outlier.

Variance and Standard Deviation: The variance and standard deviation will also be affected by the outlier, resulting in larger values due to the increased spread caused by the extreme value.